src/views/Drive.vue

```diff
@@ -783,25 +822,33 @@ module.exports = {
 			this.updateCurrentDirectory(filename);
 		},
 
-		openInApp(filename, app) {
+	    openInApp(filename, app) {
+                    if (app == null || app == "") {
+                        this.closeApps();
+                        return
+                    } else {
+                        this.closeApps(false, false);
+                    }
 		    this.selectedFiles = this.files.filter(f => f.getName() == filename);
 		    if (this.selectedFiles.length == 0)
 			return;
-                    
-		    if (app == "Gallery")
-			this.showGallery = true;
-		    else if (app == "pdf")
-			this.showPdfViewer = true;
-		    else if (app == "editor")
-			this.showCodeEditor = true;
-		    else if (app == "identity-proof")
-			this.showIdentityProof = true;
-		    else if (app == "hex")
-			this.showHexViewer = true;
-		    else if (app == "search")
-			this.showSearch = true;
-
-                    this.updateHistory(app, this.getPath, "");
+            let that = this;
+            Vue.nextTick(() => {
```

Comment:
Can you explain the need for nextTick?

Reply:
I have added comments. Required due to side-effect of history navigation